Ambiguous use of abs.

There is a stew of warnings related to use of `abs` in various contexts throughout the code-base. I assume most of them intend to reference `int abs(int)`, and will happily put together a PR to that effect, but these are pervasive enough that I would appreciate a second pair of eyes to see if any of them actually should be returning floating-point values (for example where tolerances are being checked).
```
Scanning dependencies of target buffer_static
[  3%] Building CXX object common/libBuffer/CMakeFiles/buffer_static.dir/src/CommonBuffer.cpp.o
[  3%] Building CXX object common/libBuffer/CMakeFiles/buffer_static.dir/src/CommonCharacterBuffer.cpp.o
[  4%] Building CXX object common/libBuffer/CMakeFiles/buffer_static.dir/src/CommonFWriteBufferFlusher.cpp.o
[  4%] Building CXX object common/libBuffer/CMakeFiles/buffer_static.dir/src/CommonStreamBufferFlusher.cpp.o
[  4%] Linking CXX static library ../../lib/libbuffer.a
[  4%] Built target buffer_static
[  5%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/MathMLEvaluatorVisitor.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/MathMLEvaluatorVisitor.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  5%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/MathMLSolverPrecompiled.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/MathMLSolverPrecompiled.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  5%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/MathMLSymbolTable.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/MathMLSymbolTable.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  6%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/AST/MathMLASTStringVisitor.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/AST/MathMLASTStringVisitor.cpp:2:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  6%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/AST/MathMLASTUnaryArithmeticExpression.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/AST/MathMLASTUnaryArithmeticExpression.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  6%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/AST/MathMLASTBinaryComparisionExpression.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/AST/MathMLASTBinaryComparisionExpression.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  7%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/AST/MathMLASTLogicExpression.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/AST/MathMLASTLogicExpression.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  7%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/AST/MathMLASTFragmentExpression.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/AST/MathMLASTFragmentExpression.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  7%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/AST/MathMLASTFunctionExpression.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/AST/MathMLASTFunctionExpression.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  8%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/AST/MathMLASTVariableExpression.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/AST/MathMLASTVariableExpression.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  8%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/AST/MathMLASTConstantExpression.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/AST/MathMLASTConstantExpression.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  8%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/AST/MathMLASTArithmeticExpression.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/AST/MathMLASTArithmeticExpression.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  9%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/MathMLSolverFunctionExtensions.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/MathMLSolverFunctionExtensions.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  9%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/MathMLSerializationVisitor.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/MathMLSerializationVisitor.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[  9%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/MathMLSerializationUtil.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/MathMLSerializationUtil.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[ 10%] Building CXX object Externals/MathMLSolver/CMakeFiles/MathMLSolver_static.dir/src/MathMLString.cpp.o
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/src/MathMLString.cpp:1:
In file included from /scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverStableHeaders.h:70:
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^
/scrubbedpath/OpenCOLLADA/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h:102:41: note: use function 'std::abs' instead
            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
                                        ^~~~~
                                        std::abs
1 warning generated.
[ 10%] Linking CXX static library ../../lib/libMathMLSolver.a
[ 10%] Built target MathMLSolver_static
Scanning dependencies of target zlib_static
[ 10%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/adler32.c.o
[ 10%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/compress.c.o
[ 11%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/crc32.c.o
[ 11%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/deflate.c.o
[ 11%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/gvmat32c.c.o
[ 12%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/gzio.c.o
[ 12%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/infback.c.o
[ 12%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/inffast.c.o
[ 13%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/inflate.c.o
[ 13%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/inftrees.c.o
[ 13%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/trees.c.o
[ 14%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/uncompr.c.o
[ 14%] Building C object Externals/zlib/CMakeFiles/zlib_static.dir/src/zutil.c.o
[ 15%] Linking C static library ../../lib/libzlib.a
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: ../../lib/libzlib.a(gvmat32c.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: ../../lib/libzlib.a(gvmat32c.c.o) has no symbols
[ 15%] Built target zlib_static
Scanning dependencies of target OpenCOLLADABaseUtils_static
[ 15%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/COLLADABUUtils.cpp.o
[ 16%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/COLLADABUURI.cpp.o
[ 16%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/COLLADABUPrecompiledHeaders.cpp.o
[ 16%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/COLLADABUPcreCompiledPattern.cpp.o
[ 17%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/COLLADABUIDList.cpp.o
[ 17%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/COLLADABUStringUtils.cpp.o
[ 17%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/COLLADABUHashFunctions.cpp.o
[ 18%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/COLLADABUNativeString.cpp.o
[ 18%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/Math/COLLADABUMathMatrix3.cpp.o
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:241:18: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            if ( abs( fDet ) <= fTolerance )
                 ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:241:18: note: use function 'std::abs' instead
            if ( abs( fDet ) <= fTolerance )
                 ^~~
                 std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:241:18: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:417:40: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            Real fY = kA[ 0 ][ 0 ] - ( abs( fRoot1 - fT22 ) <=
                                       ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:417:40: note: use function 'std::abs' instead
            Real fY = kA[ 0 ][ 0 ] - ( abs( fRoot1 - fT22 ) <=
                                       ^~~
                                       std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:417:40: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:418:40: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                                       abs( fRoot2 - fT22 ) ? fRoot1 : fRoot2 );
                                       ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:418:40: note: use function 'std::abs' instead
                                       abs( fRoot2 - fT22 ) ? fRoot1 : fRoot2 );
                                       ^~~
                                       std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:418:40: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:556:33: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                bool bTest1 = ( abs( kA[ 0 ][ 1 ] ) <=
                                ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:556:33: note: use function 'std::abs' instead
                bool bTest1 = ( abs( kA[ 0 ][ 1 ] ) <=
                                ^~~
                                std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:556:33: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:557:52: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                                ms_fSvdEpsilon * ( abs( kA[ 0 ][ 0 ] ) + abs( kA[ 1 ][ 1 ] ) ) );
                                                   ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:557:52: note: use function 'std::abs' instead
                                ms_fSvdEpsilon * ( abs( kA[ 0 ][ 0 ] ) + abs( kA[ 1 ][ 1 ] ) ) );
                                                   ^~~
                                                   std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:557:52: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:557:74: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                                ms_fSvdEpsilon * ( abs( kA[ 0 ][ 0 ] ) + abs( kA[ 1 ][ 1 ] ) ) );
                                                                         ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:557:74: note: use function 'std::abs' instead
                                ms_fSvdEpsilon * ( abs( kA[ 0 ][ 0 ] ) + abs( kA[ 1 ][ 1 ] ) ) );
                                                                         ^~~
                                                                         std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:557:74: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:558:33: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                bool bTest2 = ( abs( kA[ 1 ][ 2 ] ) <=
                                ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:558:33: note: use function 'std::abs' instead
                bool bTest2 = ( abs( kA[ 1 ][ 2 ] ) <=
                                ^~~
                                std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:558:33: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:559:52: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                                ms_fSvdEpsilon * ( abs( kA[ 1 ][ 1 ] ) + abs( kA[ 2 ][ 2 ] ) ) );
                                                   ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:559:52: note: use function 'std::abs' instead
                                ms_fSvdEpsilon * ( abs( kA[ 1 ][ 1 ] ) + abs( kA[ 2 ][ 2 ] ) ) );
                                                   ^~~
                                                   std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:559:52: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:559:74: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                                ms_fSvdEpsilon * ( abs( kA[ 1 ][ 1 ] ) + abs( kA[ 2 ][ 2 ] ) ) );
                                                                         ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:559:74: note: use function 'std::abs' instead
                                ms_fSvdEpsilon * ( abs( kA[ 1 ][ 1 ] ) + abs( kA[ 2 ][ 2 ] ) ) );
                                                                         ^~~
                                                                         std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:559:74: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:900:22: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                fX = abs( afCoeff[ 0 ] );
                     ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:900:22: note: use function 'std::abs' instead
                fX = abs( afCoeff[ 0 ] );
                     ^~~
                     std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:900:22: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:901:35: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                Real fTmp = 1.0 + abs( afCoeff[ 1 ] );
                                  ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:901:35: note: use function 'std::abs' instead
                Real fTmp = 1.0 + abs( afCoeff[ 1 ] );
                                  ^~~
                                  std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:901:35: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:906:30: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                fTmp = 1.0 + abs( afCoeff[ 2 ] );
                             ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:906:30: note: use function 'std::abs' instead
                fTmp = 1.0 + abs( afCoeff[ 2 ] );
                             ^~~
                             std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:906:30: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:919:22: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                if ( abs( fPoly ) <= fEpsilon )
                     ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:919:22: note: use function 'std::abs' instead
                if ( abs( fPoly ) <= fEpsilon )
                     ^~~
                     std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:919:22: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1495:18: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            if ( abs( fC ) >= EPSILON )
                 ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1495:18: note: use function 'std::abs' instead
            if ( abs( fC ) >= EPSILON )
                 ^~~
                 std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1495:18: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1552:37: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                        Real fSum = abs( afDiag[ i1 ] ) +
                                    ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1552:37: note: use function 'std::abs' instead
                        Real fSum = abs( afDiag[ i1 ] ) +
                                    ^~~
                                    std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1552:37: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1553:37: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                                    abs( afDiag[ i1 + 1 ] );
                                    ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1553:37: note: use function 'std::abs' instead
                                    abs( afDiag[ i1 + 1 ] );
                                    ^~~
                                    std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1553:37: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1555:30: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                        if ( abs( afSubDiag[ i1 ] ) + fSum == fSum )
                             ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1555:30: note: use function 'std::abs' instead
                        if ( abs( afSubDiag[ i1 ] ) + fSum == fSum )
                             ^~~
                             std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1555:30: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1582:30: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                        if ( abs( fTmp3 ) >= abs( fTmp0 ) )
                             ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1582:30: note: use function 'std::abs' instead
                        if ( abs( fTmp3 ) >= abs( fTmp0 ) )
                             ^~~
                             std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1582:30: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1582:46: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                        if ( abs( fTmp3 ) >= abs( fTmp0 ) )
                                             ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1582:46: note: use function 'std::abs' instead
                        if ( abs( fTmp3 ) >= abs( fTmp0 ) )
                                             ^~~
                                             std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp:1582:46: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
19 warnings generated.
[ 19%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/Math/COLLADABUMathVector3.cpp.o
[ 19%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/Math/COLLADABUMathUtils.cpp.o
[ 19%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/Math/COLLADABUMathQuaternion.cpp.o
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathQuaternion.cpp:365:18: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            if ( abs( fSin ) >= ms_fEpsilon )
                 ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathQuaternion.cpp:365:18: note: use function 'std::abs' instead
            if ( abs( fSin ) >= ms_fEpsilon )
                 ^~~
                 std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathQuaternion.cpp:365:18: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathQuaternion.cpp:393:18: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
            if ( abs( w ) < 1.0 )
                 ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathQuaternion.cpp:393:18: note: use function 'std::abs' instead
            if ( abs( w ) < 1.0 )
                 ^~~
                 std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathQuaternion.cpp:393:18: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathQuaternion.cpp:398:22: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                if ( abs( fSin ) >= ms_fEpsilon )
                     ^
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathQuaternion.cpp:398:22: note: use function 'std::abs' instead
                if ( abs( fSin ) >= ms_fEpsilon )
                     ^~~
                     std::abs
/scrubbedpath/OpenCOLLADA/COLLADABaseUtils/src/Math/COLLADABUMathQuaternion.cpp:398:22: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
3 warnings generated.
[ 20%] Building CXX object COLLADABaseUtils/CMakeFiles/OpenCOLLADABaseUtils_static.dir/src/Math/COLLADABUMathMatrix4.cpp.o
[ 20%] Linking CXX static library ../lib/libOpenCOLLADABaseUtils.a
```

--- a/COLLADABaseUtils/include/COLLADABUStableHeaders.h
+++ b/COLLADABaseUtils/include/COLLADABUStableHeaders.h
@@ -17,7 +17,7 @@
 #include <vector>
 #include <sstream>
 #include <algorithm>
-#include <math.h>
+#include <cmath>
 #include <string>
 #include <iostream>
 

--- a/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp
+++ b/COLLADABaseUtils/src/Math/COLLADABUMathMatrix3.cpp
@@ -238,7 +238,7 @@ namespace COLLADABU
                 m[ 0 ][ 1 ] * rkInverse[ 1 ][ 0 ] +
                 m[ 0 ][ 2 ] * rkInverse[ 2 ][ 0 ];
 
-            if ( abs( fDet ) <= fTolerance )
+            if ( std::abs( fDet ) <= fTolerance )
                 return false;
 
             Real fInvDet = 1.0 / fDet;
@@ -414,8 +414,8 @@ namespace COLLADABU
             Real fRoot2 = 0.5 * ( fTrace - fDiscr );
 
             // adjust right
-            Real fY = kA[ 0 ][ 0 ] - ( abs( fRoot1 - fT22 ) <=
-                                       abs( fRoot2 - fT22 ) ? fRoot1 : fRoot2 );
+            Real fY = kA[ 0 ][ 0 ] - ( std::abs( fRoot1 - fT22 ) <=
+                                       std::abs( fRoot2 - fT22 ) ? fRoot1 : fRoot2 );
             Real fZ = kA[ 0 ][ 1 ];
             Real fInvLength = 1/sqrt( fY * fY + fZ * fZ );
             Real fSin = fZ * fInvLength;
@@ -553,10 +553,10 @@ namespace COLLADABU
                 Real fSin0, fCos0, fTan0;
                 Real fSin1, fCos1, fTan1;
 
-                bool bTest1 = ( abs( kA[ 0 ][ 1 ] ) <=
-                                ms_fSvdEpsilon * ( abs( kA[ 0 ][ 0 ] ) + abs( kA[ 1 ][ 1 ] ) ) );
-                bool bTest2 = ( abs( kA[ 1 ][ 2 ] ) <=
-                                ms_fSvdEpsilon * ( abs( kA[ 1 ][ 1 ] ) + abs( kA[ 2 ][ 2 ] ) ) );
+                bool bTest1 = ( std::abs( kA[ 0 ][ 1 ] ) <=
+                                ms_fSvdEpsilon * ( std::abs( kA[ 0 ][ 0 ] ) + std::abs( kA[ 1 ][ 1 ] ) ) );
+                bool bTest2 = ( std::abs( kA[ 1 ][ 2 ] ) <=
+                                ms_fSvdEpsilon * ( std::abs( kA[ 1 ][ 1 ] ) + std::abs( kA[ 2 ][ 2 ] ) ) );
 
                 if ( bTest1 )
                 {
@@ -897,13 +897,13 @@ namespace COLLADABU
             if ( fPoly < 0.0 )
             {
                 // uses a matrix norm to find an upper bound on maximum root
-                fX = abs( afCoeff[ 0 ] );
-                Real fTmp = 1.0 + abs( afCoeff[ 1 ] );
+                fX = std::abs( afCoeff[ 0 ] );
+                Real fTmp = 1.0 + std::abs( afCoeff[ 1 ] );
 
                 if ( fTmp > fX )
                     fX = fTmp;
 
-                fTmp = 1.0 + abs( afCoeff[ 2 ] );
+                fTmp = 1.0 + std::abs( afCoeff[ 2 ] );
 
                 if ( fTmp > fX )
                     fX = fTmp;
@@ -916,7 +916,7 @@ namespace COLLADABU
             {
                 fPoly = afCoeff[ 0 ] + fX * ( afCoeff[ 1 ] + fX * ( afCoeff[ 2 ] + fX ) );
 
-                if ( abs( fPoly ) <= fEpsilon )
+                if ( std::abs( fPoly ) <= fEpsilon )
                     return fX;
 
                 Real fDeriv = afCoeff[ 1 ] + fX * ( fTwoC2 + 3.0 * fX );
@@ -1492,7 +1492,7 @@ namespace COLLADABU
             afDiag[ 0 ] = fA;
             afSubDiag[ 2 ] = 0.0;
 
-            if ( abs( fC ) >= EPSILON )
+            if ( std::abs( fC ) >= EPSILON )
             {
                 Real fLength = sqrt( fB * fB + fC * fC );
                 Real fInvLength = 1.0 / fLength;
@@ -1549,10 +1549,10 @@ namespace COLLADABU
 
                     for ( i1 = i0; i1 <= 1; i1++ )
                     {
-                        Real fSum = abs( afDiag[ i1 ] ) +
-                                    abs( afDiag[ i1 + 1 ] );
+                        Real fSum = std::abs( afDiag[ i1 ] ) +
+                                    std::abs( afDiag[ i1 + 1 ] );
 
-                        if ( abs( afSubDiag[ i1 ] ) + fSum == fSum )
+                        if ( std::abs( afSubDiag[ i1 ] ) + fSum == fSum )
                             break;
                     }
 
@@ -1579,7 +1579,7 @@ namespace COLLADABU
                         Real fTmp3 = fSin * afSubDiag[ i2 ];
                         Real fTmp4 = fCos * afSubDiag[ i2 ];
 
-                        if ( abs( fTmp3 ) >= abs( fTmp0 ) )
+                        if ( std::abs( fTmp3 ) >= std::abs( fTmp0 ) )
                         {
                             fCos = fTmp0 / fTmp3;
                             fTmp1 = sqrt( fCos * fCos + 1.0 );

--- a/COLLADABaseUtils/src/Math/COLLADABUMathQuaternion.cpp
+++ b/COLLADABaseUtils/src/Math/COLLADABUMathQuaternion.cpp
@@ -15,6 +15,7 @@
 #include "Math/COLLADABUMathVector3.h"
 
 
+
 namespace COLLADABU
 {
 
@@ -362,7 +363,7 @@ namespace COLLADABU
             Quaternion kResult;
             kResult.w = cos( fAngle_radian );
 
-            if ( abs( fSin ) >= ms_fEpsilon )
+            if ( std::abs( fSin ) >= ms_fEpsilon )
             {
                 Real fCoeff = fSin / fAngle_radian;
                 kResult.x = fCoeff * x;
@@ -390,12 +391,12 @@ namespace COLLADABU
             Quaternion kResult;
             kResult.w = 0.0;
 
-            if ( abs( w ) < 1.0 )
+            if ( std::abs( w ) < 1.0 )
             {
                 Real fAngle_radian ( acos( w ) );
                 Real fSin = sin( fAngle_radian );
 
-                if ( abs( fSin ) >= ms_fEpsilon )
+                if ( std::abs( fSin ) >= ms_fEpsilon )
                 {
                     Real fCoeff = fAngle_radian / fSin;
                     kResult.x = fCoeff * x;

--- a/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h
+++ b/Externals/MathMLSolver/include/MathMLSolverFunctionExtensions.h
@@ -99,7 +99,7 @@ namespace MathML
         //-------------------------------------------------------------------------------------------
         inline static void abs( MathML::AST::ConstantExpression& result, const ScalarList& paramlist, ErrorHandler* errorHandler )
         {
-            result.setValue( ( double ) ::abs( paramlist.at( 0 ).getDoubleValue() ) );
+            result.setValue( ( double ) std::abs( paramlist.at( 0 ).getDoubleValue() ) );
         }
 
         //-------------------------------------------------------------------------------------------

